PA193_mnemonic_Slytherin/mnemonic.py

```diff
@@ -97,17 +99,28 @@ def recover(mnemonic: str, seed_password: str = '') -> Tuple[bytes, bytes]:
     :rtype: Tuple[bytes, bytes]
     :return: Two item tuple where first is initial entropy and second is seed.
     """
-    pass
+    if not __is_valid_mnemonic(mnemonic):
+        raise ValueError('invalid mnemonic')
+
+    entropy = __mnemonic2entropy(mnemonic)
+    seed = __generate_seed(mnemonic, seed_password)
+    return entropy, seed
 
 
-def verify(mnemonic_phrase: str, expected_seed: bytes, seed_password: str = '') -> bool:
+def verify(mnemonic: str, expected_seed: bytes, seed_password: str = '') -> bool:
     """Verify whether mnemonic phrase matches with expected seed.
     Seed can be protected by password. If a seed should not be protected, the password is treated as `''`
     (empty string) by default.
     :rtype: bool
     :return: True if provided phrase generates expected seed, False otherwise.
     """
-    pass
+    if not __is_valid_mnemonic(mnemonic):
+        raise ValueError('invalid mnemonic')
+    if not __is_valid_seed(expected_seed):
+        raise ValueError('invalid expected_seed')
+
+    entropy, generated_seed = recover(mnemonic, seed_password)
```

Comment:
Why do we need to compute entropy in the verifying function?

Reply:
We can only generate seed, sure. :)